GGRC-6081 Fix review status change on folder update

aalekseev opened the pull request:
# Issue description

Updating a folder field on any folderable model resets review status to "Unreviewed". The desired result is that review status will preserve through updates on folder field.

# Steps to test the changes

1. Create control
2. Mark Reviewed
3. Assign Folder

Expected result: control state -> Reviewed

# Solution description

Review model checks changed fields and if there is a field not listed in ATTRS_TO_IGNORE set, then review status automatically switches to "Unreviewed". Extending that set to include "folder" field resolves the issue.

# Sanity checklist

- [x] I have clicked through the app to make sure my changes work and not break the app.
- [x] I have applied the correct milestone and labels.
- [x] My changes fix the issue described in the description (and do nothing else). 🤞
- [x] My changes are covered by tests.
- [ ] My changes follow our [performance guidelines](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/performance.rst).
- [x] My changes follow our [js](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/javascript.rst) and/or [python](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/python.rst) guidelines.
- [x] My commits follow our [commit guidelines](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/git/how_to_write_a_commit_message.rst).

<!-- If your PR includes a migration include the additional checklist items
# Migration checklist
- [ ] Migration passes all checks from our [PR review guidelines](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/git/reviewing_pull_requests.rst#reviewing-a-pr-containing-database-migration-scripts)
-->

# PR Review checklist

- [ ] The changes fix the issue and don't cause any apparent regressions.
- [ ] Labels and milestone are correctly set.
- [ ] The solution description matches the changes in the code.
- [ ] There is no apparent way to improve the performance & design of the new code.
- [ ] The pull request is opened against the correct base branch.
- [ ] Upon merging, the Jira ticket's fixversion is correctly set and the ticket is moved to "QA - In Progress".

<!-- If your code is not finished yet can include a TODO check list
# TODO

- [ ] First item on the TODO
-->

aalekseev commented:
I signed it!

olegnatsevsky commented:
@aalekseev Congratulations with new contribution! PR looks good except 2 minor things:
1) code style need to be fixed.
```
 Initializing environment
 ./test/integration/ggrc/models/test_review.py:23:7: E121 continuation line under-indented for hanging indent
 ./test/integration/ggrc/converters/test_import_reviewable.py:394:7: E121 continuation line under-indented for hanging indent
 ./test/integration/ggrc/converters/test_import_reviewable.py:398:7: E121 continuation line under-indented for hanging indent
 ./test/integration/ggrc/converters/test_import_reviewable.py:399:9: E121 continuation line under-indented for hanging indent
 ./test/integration/ggrc/converters/test_import_reviewable.py:408:7: E121 continuation line under-indented for hanging indent

 Pylint results.
 Number of issues on old commit: 0 
 Number of issues on current commit: 1

```
2) test location.
we have tests for reviewable status update in 
```
integration/ggrc/review/test_status_auto_update.py
```
please relocate your tests there.

aalekseev commented:
@olegnatsevsky, thank you for the feedback!

1. Updated local branch from dev
2. Checked code style
3. Moved tests
4. Updated commit message
5. Refactored duplicated code in review tests

olegnatsevsky commented on test/integration/ggrc/review/__init__.py:
```diff
@@ -22,3 +23,34 @@ def build_reviewer_acl(acr_id=None, user_id=None):
   }]
+
+
+def generate_review_object(
```
GJ! Thanks for improvement! But we can go even further. Review can be attached to different objects not only control. Let's upgrade this helper to more generic.
you can use factory.get_model_factory(model_name) to load object by model name.

aalekseev replied:
Thanks for the tip!
I changed function declaration from control to instance. Maybe generating not only review object but multiple will be too much for this function?
